### PR TITLE
[Resubmit] Require inductor to match stride order

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91185

Resubmitting https://github.com/pytorch/pytorch/pull/90563 because I had a commit in that stack which didn't use my CLA-approved git username. 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by desertfire — APPROVED** [review]

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -273,22 +274,22 @@ def process(device_type):
     "corrcoef": {f16, f32, f64, i32, i64},
     "cov": {f16, f32, f64, i32, i64},
     "equal": {b8, f16, f32, f64, i32, i64},
-    "fft.fft": {f16, f32, f64},
+    "fft.fft": {b8, f16, f32, f64, i32, i64},
```
Comment (desertfire) [inline test/inductor/test_torchinductor_opinfo.py:277]: What's the story about these new expected failures?

Comment (eellison) [reply test/inductor/test_torchinductor_opinfo.py:277]: fft support here is extremely flakey. these kernels output complex, so the `requiring_stride_order` on the end breaks when we interact with complex dtypes

File: torch/_inductor/graph.py [hunk 2]
```diff
@@ -366,6 +366,20 @@ def run_node(self, n: torch.fx.Node):
             else:
                 result = super().run_node(n)
 
+            # require the same stride order for dense outputs,
+            # so that user-land view() will not throw because inductor
+            # output different strides than eager
+            # long term the solution is to make view() always succeed
+            # with infallible strides.
```
Comment (anijain2305) [inline torch/_inductor/graph.py:373]: Just for understand (the PR is ok) - What is view with infallible strides?

**Review by anijain2305 — APPROVED** [review]

File: torch/_inductor/graph.py (hunk 2 above)
Comment (eellison) [reply torch/_inductor/graph.py:373]: https://docs.google.com/document/d/1wWUd_hBoCwnMV2CnJNWxNFMRZYFyspV_iyI_1Nya7c0/edit#heading=h.lmmok2ikj8ad

**Pull request closed without merging**